src/components/PageExport/PageExport.vue

```diff
@@ -0,0 +1,163 @@
+        </v-row>
+      </Fragment>
+    </div>
+  </section>
+</template>
+
+<script>
+  import JSZip from 'jszip'
+  export default {
+    props: {
+      isLoading: {
+        type: Boolean,
+        default: false,
+      },
+      reservoirs: {
+        type: Array,
+        default: () => [],
+      },
+      timeSeries: {
+        type: [Object, null],
+        default: null,
+      },
+      areaType: {
+        type: String,
+        default: '',
+      },
+    },
+
+    data: () => ({
+      shareUrl: undefined,
+    }),
+
+    mounted () {
+      this.shareUrl = window.location.href
+    },
+
+    methods: {
+      async copyToClipBoard () {
+        try {
+          await navigator.clipboard.writeText(this.shareUrl)
+        } catch (err) {
+          throw new Error('Error: Copy to clipboard')
+        }
+      },
+      async exportTimeSeries () {
+        if (this.timeSeries.series.length === 1) {
+          const { name, content } = this.timeSeriesCsv(this.timeSeries.series[0])
+
+          this.downloadFile(
+            `${name}.csv`,
+            new Blob([content], { type: 'text/csv;charset=utf-8' }),
+          )
+        } else {
+          const zip = this.timeSeriesZip()
+
+          const content = await zip.generateAsync({ type: 'blob' })
+          this.downloadFile(
+            'Reservoirs - Surface Water Area.zip',
+            content,
+          )
+        }
+      },
+      exportGeometry () {
+        if (this.reservoirs.length === 1) {
+          const geometry = JSON.stringify(this.reservoirs[0])
+
+          this.downloadFile(
+            this.reservoirs[0]?.properties?.name
+              ? `${this.reservoirs[0].properties.name} (#${this.reservoirs[0].id}).geojson`
+              : `#${this.reservoirs[0].id}.geojson`,
+            new Blob([geometry], { type: 'application/geo+json;charset=utf-8' }),
+          )
+        } else {
+          const geometry = JSON.stringify({
+            type: 'FeatureCollection',
+            features: this.reservoirs,
+          })
+
+          this.downloadFile(
+            'Reservoirs.geojson',
+            new Blob([geometry], { type: 'application/geo+json;charset=utf-8' }),
+          )
+        }
+      },
+      timeSeriesCsv (serie) {
+        let csv = `${this.timeSeries.xAxis[0].type},${this.timeSeries.yAxis[0].name}\n`
+        serie.data.forEach((row) => {
+          csv += row.join(',')
+          csv += '\n'
+        })
+        return { name: serie.name, content: csv }
+      },
+      timeSeriesZip () {
+        const zip = new JSZip()
+        const files = this.timeSeries.series.map((serie) => {
+          return this.timeSeriesCsv(serie)
+        })
+        files.forEach(({ name, content }) => {
+          zip.file(`${name}.csv`, content)
+        })
+        return zip
+      },
+      downloadFile (filename, blob) {
+        const anchor = document.createElement('a')
+        anchor.href = (window.URL || window.webkitURL).createObjectURL(blob)
```

Comment:
why is webkitURL needed here?

Reply:
Firefox uses `window.URL` and Chrome `window.webkitURL`, it's common practice to write it like this